### PR TITLE
Add option for preserving non-F strings when joining

### PR DESCRIPTION
**Pull request by amosjyng**
Currently, whenever I join pure strings, the whole thing gets highlighted. I would prefer for there to be an option where this is not the case.

If you're okay with modifying the existing `join` function, I can certainly try to change that instead.

### CODE REVIEW

File: tests/test_f.py [hunk 1]
```diff
@@ -288,6 +288,51 @@ def test_join_bad_source():
...
+
+
+def test_preserved_join_list():
+    b = 2
+    s = F(" ").preserved_join(["a", F(f"b={b}"), "c"])
+    assert s == "a b=2 c"
+    assert s.parts == (
+        "a",
+        " ",
+        "b=",
```
Comment (alexmojaki) [inline tests/test_f.py:303]: I think it makes sense to want to be able to get to these literal parts and not be stuck with things like `FValue(source="(strings)[1]", value="b", formatted="b")`, but I think that should only happen after flattening, and this should probably still be `F(f"b={b}")`.

Comment (alexmojaki) [reply tests/test_f.py:303]: (or an FValue containing that)

Comment (amosjyng) [reply tests/test_f.py]: Ahh I see what you mean. Ok done!

**Review by alexmojaki — APPROVED** [review]
Sorry for the silence, been overwhelmed lately. LGTM, thanks!

**Comment by amosjyng** [comment]
No worries, I've been using this code privately anyways. I just figured it might be useful to others in the base library as well :)

Would you like to merge the PR? I don't have permission to do that.

Thanks for the review!

**Comment by alexmojaki** [comment]
Released 0.0.4 on PyPI.

**Pull request merged**